File: src/main/java/com/github/kuro46/embedscript/request/Request.java
package com.github.kuro46.embedscript.request;

import com.github.kuro46.embedscript.script.Script;

/**
 * @author shirokuro
 */
public class Request {
    private final RequestType requestType;
    private final Script script;

    public Request(RequestType requestType) {
        this(requestType, null);
    }

    public Request(RequestType requestType, Script script) {
        this.requestType = requestType;
        this.script = script;
    }

    public RequestType getRequestType() {
        return requestType;
    }

    public Script getScript() {
        return script;
    }
}

File: src/main/kotlin/com/github/kuro46/embedscript/request/Request.kt
package com.github.kuro46.embedscript.request

import com.github.kuro46.embedscript.script.Script

/**
 * @author shirokuro
 */
sealed class Request {
    data class Embed(val scripts: List<Script>) : Request() {
        constructor(script: Script) : this(listOf(script))
    }
    data class Add(val scripts: List<Script>) : Request() {
        constructor(script: Script) : this(listOf(script))
    }
    object Remove : Request()
    object View : Request()
}
